tt_metal/llrt/hal.hpp

```diff
@@ -41,10 +41,22 @@ enum class HalMemAddrType : uint8_t {
     COUNT = 8
 };
 
+enum class HalMemType : uint8_t {
+    L1 = 0,
+    DRAM = 1,
+    HOST = 2,
+    COUNT = 3
+};
+
 using DeviceAddr = std::uint64_t;
 
 class Hal;
 
+template <typename E, std::enable_if_t<std::is_enum<E>::value, bool> = true>
```

Comment:
ideally this would go in a common "misc" header file

Reply:
will move to common/utils.hpp